node/flatpak_node_generator/providers/npm.py

```diff
@@ -104,20 +111,15 @@ def _process_packages_v2(
                 # NOTE We're not interested in symlinks, NPM will create them at install time
                 # but we still could collect package symlinks anyway just for completeness
                 continue
+            if install_path == '':
+                # The root project is typically listed with a key of ''
+                continue
 
             name = info.get('name')
 
             source: PackageSource
-            package_json_path = lockfile.parent / install_path / 'package.json'
-            if (
```

Comment:
what's the reason behind this change? if "entry without a `resolved` is always a local path" is reliable, then yeah it would make removing this hack doable...but that would ideally be in a separate commit before this one, to be able to tell the logic apart more easily.

Reply:
Agreed, when we've worked through all other suggestions to satisfaction I'll pull this out to its own commit and rebase the branch again ready for final merge.